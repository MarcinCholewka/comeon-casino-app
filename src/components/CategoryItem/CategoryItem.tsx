import { List } from 'semantic-ui-react';

import { TCategory } from '@components/Categories';
import { useGamesFilters } from '@hooks/useGamesFilterContext';

type Props = {
  category: TCategory;
};

export const CategoryItem = ({ category }: Props) => {
  const { id, name } = category;
  const { setCategory, filters } = useGamesFilters();
  const activeCategory = filters.category === id;

  const handleCategoryClick = () => setCategory(id);

  return (
    <List.Item active={activeCategory} onClick={handleCategoryClick}>
      {name}
    </List.Item>
  );
};
