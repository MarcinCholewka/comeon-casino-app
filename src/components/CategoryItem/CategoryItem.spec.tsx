import { render } from '@testing-library/react';
import { CategoryItem } from './CategoryItem';

describe('<CategoryItem />', () => {
  const category = {
    id: 1,
    name: 'Category #1',
  };

  it('should render without crashing', () => {
    const { container, getByText } = render(
      <CategoryItem category={category} />,
    );
    const categoryName = getByText(category.name);

    expect(categoryName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
