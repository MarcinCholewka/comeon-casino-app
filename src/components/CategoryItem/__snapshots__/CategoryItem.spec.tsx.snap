// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> should render without crashing 1`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    Category #1
  </div>
</div>
`;
